cmd/ova-provider-server/ova-provider-server.go

```diff
@@ -453,3 +500,37 @@ func getDiskPath(path string) string {
 	}
 	return path
 }
+
+type UUIDMap struct {
+	m map[string]string
+}
+
+func NewUUIDMap() *UUIDMap {
+	return &UUIDMap{
+		m: make(map[string]string),
+	}
+}
+
+func (um *UUIDMap) GetUUID(filename string) string {
+	var id string
+	id, ok := um.m[filename]
+
+	if !ok {
+		uuid := make([]byte, 16)
+		n, err := io.ReadFull(rand.Reader, uuid)
+		if n != len(uuid) || err != nil {
+			return ""
+		}
+		uuid[6] = (uuid[6] & 0x0f) | 0x40
+		uuid[8] = (uuid[8] & 0x3f) | 0x80
+		id = fmt.Sprintf("%x-%x-%x-%x-%x", uuid[0:4], uuid[4:6], uuid[6:8], uuid[8:10], uuid[10:])
+
+		um.m[filename] = id
+	}
+	return id
+}
+
+func isValidUUID(id string) bool {
```

Comment:
did you look into using github.com/google/uuid for validating and generating UIDs?

Reply:
yes, I've initially used it, but it didn't work in the pod, probably need to add download in the container file i"ll look into that again

Reply:
I just remembered that we were planning to use the vm hash as an ID (for consistency if the server will crush), changing in the code

Reply:
ok, that applies to generating the UUID but we can still validate UIDs with uuid.Parse